File: Commons/AOP Tests/src/test/java/com/greentree/commons/tests/aop/TestConfig.java
package com.greentree.commons.tests.aop;

public class TestConfig {

    @AutowiredProvider(tags = "Anton")
    Person personAnton() {
        return Anton.INSTANCE;
    }

    @AutowiredProvider
    Shape newRect() {
        return new Rect(15f);
    }

    @AutowiredProvider
    Shape newTriangle() {
        return new Triangle(15f);
    }

    //    @AutowiredProvider
//    Person personCopyWithLastName(Person person) {
//        return new NamePerson(person.getName() + " AAA");
//    }
//
    @AutowiredProvider
    Person personFromName(Name name) {
        return new NamePerson(name);
    }

    @AutowiredProvider(tags = "Arseny")
    Name nameArseny() {
        return Arseny.INSTANCE;
    }

}

File: Commons/Context/Tests/src/test/java/aop/TestConfig.kt
package aop

import com.greentree.commons.context.MutableBeanContext
import com.greentree.commons.context.layer.BeanLayer
import com.greentree.commons.context.registerInstance
import com.greentree.commons.context.resolveAllBeans

class TestConfig : BeanLayer {

	override fun MutableBeanContext.register() {
		registerInstance(Arseny)
		registerInstance(Anton)
		registerInstance(Rect(15f))
		registerInstance(Triangle(15f))
		for(name in resolveAllBeans<Name>().toList()) {
			registerInstance(NamePerson(name))
		}
	}
}
